pyhik/hikvision.py

```diff
@@ -430,6 +432,182 @@ def get_device_info(self):
             _LOGGING.error('There was a problem: %s', err)
             return None
 
+    """
+    This function is added by Ayush Pratap Singh (ayushs56@gmail.com)
+    This function is used to get the status of the device
+    """
+
+    def get_device_status(self):
+        """Parse deviceInfo into dictionary."""
+        device_status = {}
+        url = '%s/ISAPI/System/status' % self.root_url
+        using_digest = False
+
+        try:
+            response = self.hik_request.get(url, timeout=CONNECT_TIMEOUT)
+            if response.status_code == requests.codes.unauthorized:
+                _LOGGING.debug('Basic authentication failed. Using digest.')
+                self.hik_request.auth = HTTPDigestAuth(self.usr, self.pwd)
+                using_digest = True
+                response = self.hik_request.get(url)
+
+            if response.status_code == requests.codes.not_found:
+                # Try alternate URL for deviceInfo
+                _LOGGING.debug('Using alternate deviceInfo URL.')
+                url = '%s/System/status' % self.root_url
+                response = self.hik_request.get(url)
+                # Seems to be difference between camera and nvr, they can't seem to
+                # agree if they should 404 or 401 first
+                if not using_digest and response.status_code == requests.codes.unauthorized:
+                    _LOGGING.debug('Basic authentication failed. Using digest.')
+                    self.hik_request.auth = HTTPDigestAuth(self.usr, self.pwd)
+                    using_digest = True
+                    response = self.hik_request.get(url)
+
+        except (requests.exceptions.RequestException,
+                requests.exceptions.ConnectionError) as err:
+            _LOGGING.error('Unable to fetch status of device, error: %s', err)
+            return None
+
+        if response.status_code == requests.codes.unauthorized:
+            _LOGGING.error('Authentication failed')
+            return None
+
+        if response.status_code != requests.codes.ok:
+            # If we didn't receive 200, abort
+            _LOGGING.debug('Unable to fetch status of device.')
+            return None
+
+        try:
+            tree = ET.fromstring(response.text)
+
+            device_status['currentdevicetime'] = tree[0].text.strip()
```

Comment:
Use the `element_query` function to find the fields you want.  This allows the XML ordering to change and you can still return the correct result.